hazelcast/src/main/java/com/hazelcast/config/CacheConfigReadOnly.java

```diff
@@ -75,124 +75,128 @@ public String getQuorumName() {
-        throw new UnsupportedOperationException("This config is read-only cache: " + getName());
+        throw throwReadOnly();
     }
 
     @Override
     public CacheConfig<K, V> setWanReplicationRef(final WanReplicationRef wanReplicationRef) {
-        throw new UnsupportedOperationException("This config is read-only cache: " + getName());
+        throw throwReadOnly();
     }
 
     @Override
     public CacheConfig<K, V> setQuorumName(String quorumName) {
-        throw new UnsupportedOperationException("This config is read-only cache: " + getName());
+        throw throwReadOnly();
     }
 
     @Override
     public CacheConfiguration<K, V> setHotRestartConfig(HotRestartConfig hotRestartConfig) {
-        throw new UnsupportedOperationException("This config is read-only cache: " + getName());
+        throw throwReadOnly();
     }
 
     @Override
     public CacheConfig<K, V> setPartitionLostListenerConfigs(
             List<CachePartitionLostListenerConfig> partitionLostListenerConfigs) {
-        throw new UnsupportedOperationException("This config is read-only cache: " + getName());
+        throw throwReadOnly();
     }
 
     @Override
     public void setMergePolicy(String mergePolicy) {
-        throw new UnsupportedOperationException("This config is read-only cache: " + getName());
+        throw throwReadOnly();
     }
 
     @Override
     public CacheConfiguration<K, V> setExpiryPolicyFactory(Factory<? extends ExpiryPolicy> expiryPolicyFactory) {
-        throw new UnsupportedOperationException("This config is read-only cache: " + getName());
+        throw throwReadOnly();
     }
 
     @Override
     public CacheConfiguration<K, V> setCacheLoaderFactory(Factory<? extends CacheLoader<K, V>> cacheLoaderFactory) {
-        throw new UnsupportedOperationException("This config is read-only cache: " + getName());
+        throw throwReadOnly();
     }
 
     @Override
     public CacheConfiguration<K, V> setCacheWriterFactory(
             Factory<? extends CacheWriter<? super K, ? super V>> cacheWriterFactory) {
-        throw new UnsupportedOperationException("This config is read-only cache: " + getName());
+        throw throwReadOnly();
     }
 
     @Override
     public CacheConfiguration<K, V> setWriteThrough(boolean isWriteThrough) {
-        throw new UnsupportedOperationException("This config is read-only cache: " + getName());
+        throw throwReadOnly();
     }
 
     @Override
     public CacheConfiguration<K, V> setReadThrough(boolean isReadThrough) {
-        throw new UnsupportedOperationException("This config is read-only cache: " + getName());
+        throw throwReadOnly();
     }
 
     @Override
     public void setDisablePerEntryInvalidationEvents(boolean disablePerEntryInvalidationEvents) {
+        throw throwReadOnly();
+    }
+
+    private UnsupportedOperationException throwReadOnly() {
         throw new UnsupportedOperationException("This config is read-only cache: " + getName());
```

Comment:
Maybe `This is a read only cache config` or simply `This config is read-only`. This message doesn't make much sense. Same for others.

Reply:
I think `This config is read-only` is the shortest and best message. We should use it in all our read-only configs (until we hopefully get rid of them in HZ 4.0). A user should anyway never see it, since I doubt anyone beside ever uses the `getAsReadOnly()` methods.